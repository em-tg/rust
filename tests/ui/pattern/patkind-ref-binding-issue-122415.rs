//@ check-pass
//@ run-rustfix
#![allow(dead_code)]

fn mutate(_y: &mut i32) {}

fn foo(&x: &i32) {
    mutate(&mut x);
    //~^ WARNING cannot borrow `x` as mutable
}

fn main() {}
