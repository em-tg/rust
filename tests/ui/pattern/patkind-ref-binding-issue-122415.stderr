warning[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
  --> $DIR/patkind-ref-binding-issue-122415.rs:8:12
   |
LL |     mutate(&mut x);
   |            ^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL | fn foo(&(mut x): &i32) {
   |        ~~~~~  +

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
